File: src/Core/Chms.Domain/Entities/Member.cs
using Chms.Domain.Common.Enums;

namespace Chms.Domain.Entities;

    public class Member : BaseModel
    {
        public Guid Id { get; set; }
        public string? FirstName { get; set; }
        public string? MiddleName {get; set;}
        public string? LastName { get; set; }
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public string? SecondaryPhoneNumber { get; set;}
        public string? BirthDate { get; set; }
        public string? Gender{ get; set;}
        public string? Occupation { get; set; }
        public string? Photo { get; set; }
        public string? PermanentAddress { get; set; }
        public string? TemporaryAddress { get; set; }
        public int GroupId { get; set; } 
        public ChurchRole ChurchRole { get; set; }

        public string? MembershipDate {get; set;}
        public string? BaptizedDate {get; set;}
            
    }

    public class MemberListVM : BaseModel
    {
        public Guid Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? MiddleName { get; set; }
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public string? Gender{ get; set;}
        public int GroupId { get; set; } 
        public string? Photo {get; set;}
        public ChurchRole ChurchRole { get; set; }
    }

File: API/app/Models/Member.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Member extends Model
{
    /** @use HasFactory<\Database\Factories\MemberFactory> */
    use HasFactory;

    protected $table = 'members';
    protected $primaryKey = 'id';
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'id',
        'first_name',
        'middle_name',
        'last_name',
        'gender',
        'email',
        'phone_number',
        'secondary_phone_number',
        'birth_date',
        'occupation',
        'photo',
        'permanent_address',
        'temporary_address',
        'baptized_date',
        'membership_date',
        'group_id',
        'church_role',
        'created_by',
        'updated_by',
    ];

    // Automatically set UUID on creating
    protected static function boot()
    {
        parent::boot();

        static::creating(function ($model) {
            if (empty($model->{$model->getKeyName()})) {
                $model->{$model->getKeyName()} = (string) Str::uuid();
            }
        });
    }
}
